### DISCUSSION TITLE
External tool to replace plain HTTP(S) links with the corresponding page title

### DISCUSSION THREAD

**Discussion by naquad** [zim-desktop-wiki/zim-desktop-wiki#2536] [Show and tell]
Hi,

I've got lots of pages containing HTTP(S) links. Like dozens of them and that didn't look good. I've been inspired by the Obsidian plugin replacing plain links with the corresponding HTML title. The result is not always nice, but definitely more human-readable + editing the title later is easier (doesn't require a dialogue).

[Link to the Gist](https://gist.github.com/naquad/19df987e70ac7c11e9d173bed22ed6aa)

Add it as an external tool with `%f` parameter.

Highlights:
* Parallel requests
* Doesn't screw up failed links
* Doesn't touch the literal blocks & strings

Requirements:
* BeautifulSoup4
* requests

**Comment by lorenrosen** [comment]
Doesn't help with old links, but I've been using the Create Link extension to Chrome to create a link with the page title of the linked to page.

**Reply by naquad** [reply]
I hit `ctrl+l alt+tab ctrl+v` before I can even think of what I'm doing, so lots and lots of plain links :( But thanks for the info, [that's the extension](https://chromewebstore.google.com/detail/create-link/gcmghdmnkfdbncmnmlkkglmnnhagajbm?pli=1) as I understand.
